housePaint.java

```diff
@@ -0,0 +1,28 @@
+/**Time Complexity :
+    O(n)
+// Space Complexity :
+    we are using same matrix hence O(n)
```

Comment:
Since we are using same matrix , so there is no auxiliary space used . So space would be O(1)